frontend/src/pages/petlistings/PetListingsPage.jsx

```diff
@@ -0,0 +1,90 @@
+  const [page, setPage] = useState(1)
+  
+  const [query, setQuery] = useState('')
+  const [endPoint, setEndPoint] = useState('')
+  
+  const { data } = useFetch(endPoint, {
+    headers: {
+      Authorization: `Bearer ${token}`
+    }
+  })
+
+  useEffect(() => {
+    const searchTerm = searchParams.get('search')
+    setSearchTerm(searchTerm)
+  }, [searchParams])
+
+  // Get complete query string
+  useEffect(() => {
+    let queryString = '';
+
+    if (Object.keys(filters).length) {
+      let queryFilter = new URLSearchParams(filters).toString()
+      queryString = queryString + queryFilter
+    }
+
+    if (searchTerm) {
+      const parseSearchArr = searchTerm.split(' ')
+      let name = ''
+      for (let i in parseSearchArr) {
+        if(parseSearchArr[i] === 'dog') queryString = queryString + (queryString !== '' ? '&category=D' : 'category=D')
+        else if(parseSearchArr[i] === 'cat') queryString = queryString + (queryString !== '' ? '&category=C' : 'category=C')
+        else if(parseSearchArr[i] === 'other') queryString = queryString + (queryString !== '' ? '&category=O' : 'category=O')
+        else if(parseSearchArr[i] === 'small') queryString = queryString + (queryString !== '' ? '&size=S' : 'size=S')
+        else if(parseSearchArr[i] === 'medium') queryString = queryString + (queryString !== '' ? '&size=M' : 'size=M')
+        else if(parseSearchArr[i] === 'large') queryString = queryString + (queryString !== '' ? '&size=L' : 'size=L')
+        else if(parseSearchArr[i] === 'male') queryString = queryString + (queryString !== '' ? '&gender=M' : 'gender=M')
+        else if(parseSearchArr[i] === 'female') queryString = queryString + (queryString !== '' ? '&gender=L' : 'gender=L')
+        else if(parseSearchArr[i] === 'available') queryString = queryString + (queryString !== '' ? '&status=AV' : 'status=AV')
+        else if(parseSearchArr[i] === 'adopted') queryString = queryString + (queryString !== '' ? '&status=AD' : 'status=AD')
+        else if(parseSearchArr[i] === 'pending') queryString = queryString + (queryString !== '' ? '&status=PE' : 'status=PE')
+        else if(parseSearchArr[i] === 'withdrawn') queryString = queryString + (queryString !== '' ? '&status=WI' : 'status=WI')
+        else if(parseSearchArr[i].includes('@')) queryString = queryString + (queryString !== '' ? `&shelter_email=${parseSearchArr[i]}` : `shelter_email=${parseSearchArr[i]}`)
+        else name = name + (name === '' ? parseSearchArr[i] : ("%20" + parseSearchArr[i]))
+      }
+      if(name !== '') queryString = queryString + (queryString !== '' ? `&name=${name}` : `name=${name}`)
+    }
+
+    if (sortOption && sortOption !== "none") queryString = queryString + (queryString !== '' ? `&sort_by=${sortOption}&ascending=${sortAsc}` : `sort_by=${sortOption}&ascending=${sortAsc}`)
+    if (page > 1) queryString = queryString + (queryString !== '' ? `&page=${page}`:`page=${page}`)
+  
+    setQuery(queryString)
+    console.log(queryString)
```

Comment:
remove

Reply:
Fixed in [force-push](https://github.com/lujorryn/PetPal-G0346/compare/85a1a86f83af31181ba802c724ed58323bc5967c..bdacb5149e4e6357c93a8983e28f28f4d937f2df)